pkg/app/piped/platformprovider/kubernetes/kustomize.go

```diff
@@ -37,12 +37,19 @@ func NewKustomize(version, path string, logger *zap.Logger) *Kustomize {
 	}
 }
 
-func (c *Kustomize) Template(ctx context.Context, appName, appDir string, opts map[string]string) (string, error) {
+func (c *Kustomize) Template(ctx context.Context, appName, appDir string, opts map[string]string, helm *Helm) (string, error) {
 	args := []string{
 		"build",
 		".",
 	}
 
+	// Pass the Helm command path to kustomize to use the specified version of Helm.
+	// Unconditionally adding this flag as it's unharmful when Helm is not used.
+	// Note: It's only available on Kustomize v4.1.0 and higher.
+	if c.version >= "4.1.0" && helm != nil {
```

Comment:
I apologize for overlooking this version comparison issue.
We cannot compare semantic versions by simply comparing strings. For instance, versions greater than or equal to 10 are incorrectly evaluated as less than 9.
Could you please implement a feature to compare semantic versions and include corresponding tests?

Reply:
Ah, right.
Fixed and added tests in [`84bfe4d` (#5636)](https://github.com/pipe-cd/pipecd/pull/5636/commits/84bfe4db299c1cdae3621608acda94b83da78db5)